### PR TITLE
Add some useful methods to NodeList

### LINKED ISSUE #1183: Add FieldDeclaration to top of ClassOrInterfaceDeclaration
How can I add a FieldDeclaration that's the first member of the ClassOrInterfaceDeclaration?

I can't find any easy way of doing this.

### PR DESCRIPTION
**Pull request by matozoid**
Fixes #1183

### CODE REVIEW

File: javaparser-core/src/main/java/com/github/javaparser/ast/NodeList.java [hunk 1]
```diff
@@ -165,6 +165,51 @@ public void add(int index, N node) {
         innerList.add(index, node);
     }
 
+    /**
+     * Inserts the node before all other nodes.
+     */
+    public NodeList<N> addToTop(N node) {
```
Comment (ftomassetti) [inline javaparser-core/src/main/java/com/github/javaparser/ast/NodeList.java]: What about `addAsFirst`? Or `addAtTheBeginning`?

File: javaparser-core/src/main/java/com/github/javaparser/ast/NodeList.java [hunk 2]
```diff
@@ -165,6 +165,51 @@ public void add(int index, N node) {
...
+        add(0, node);
+        return this;
+    }
+
+    /**
+     * Inserts the node after all other nodes. (This is simply an alias for add.)
+     */
+    public NodeList<N> addToBottom(N node) {
```
Comment (ftomassetti) [inline javaparser-core/src/main/java/com/github/javaparser/ast/NodeList.java]: What about `addAsLast`? or `addAtTheEnd`?

**Review by ftomassetti — APPROVED** [review]

File: javaparser-core/src/main/java/com/github/javaparser/ast/NodeList.java (hunk 1 above)
Comment (matozoid) [reply javaparser-core/src/main/java/com/github/javaparser/ast/NodeList.java]: I took the naming of the related issue. I'm not sure what to take as a good source for naming these - what do list implementation normally use? And other node list implementations? What do parser people say?

**Comment by DeepSnowNeeL** [comment]
addFirst
Inserts the specified element at the beginning of this list.
addLast
Appends the specified element to the end of this list.

https://docs.oracle.com/javase/7/docs/api/java/util/LinkedList.html

**Comment by matozoid** [comment]
@DeepSnowNeeL nice, thanks!

**Comment by matozoid** [comment]
@ftomassetti @DeepSnowNeeL is it better like this?

**Pull request merged**